src/NodeHelper.groovy

```diff
@@ -1213,4 +1213,18 @@ class  NodeHelper {
 
         return ret;
     }
+
+    /**
+     * Determine if a node exists and is online for the given node label
+     *
+     * @param nodeLabel
+     * @return
+     */
+    public static boolean nodeIsOnline(String nodeLabel) {
+        return Jenkins.getInstance().getLabel(nodeLabel).nodes
```

Comment:
How does it behave when you pass a non existent label, does it crash since it will calling .nodes on null?

Reply:
I havnt tested this but my guess is it will return a label with no nodes as it appears that Jenkins does not have a concept of labels that do not exist

Reply:
yeah just tested it, works fine

Reply:
what I meant was what if user passes a label that doesn't exist. But it's great that it handles it! Good work!

Reply:
Any label you ask for exists, as in there is no concept of a label that does not exist, only labels that have no nodes. i.e:

https://ci.adoptopenjdk.net/label/i-am-a-label-that-does-not-exist-but-i-do/

Reply:
ah, you are right! my apologies for that misunderstanding